### ISSUE TITLE
High CPU in webview2 when opening large file

### ISSUE THREAD

**Issue by greatquux** [WaynePhillipsEA/twinbasic#833]
When opening a large file (in this case 4000+ lines) there is a long delay while the color coding is applied.  The file opens in B&W at first and when the color coding process is finished the CPU goes down again.  This took several minutes on a 3.3GHz Core i5.  Also kind of impossible to work with the file since modifications trigger it again.  This is WebView2 version 101.0.1210.39.  Not sure if there are any GPU acceleration processes or something I can try to change?

![image](https://user-images.githubusercontent.com/1655942/167714513-9927d3fb-a60e-46f9-9c53-d41fea260804.png)

**Comment by WaynePhillipsEA** [comment]
I'm hopeful for a little improvement here in BETA 33.  Let me know how it goes for you.

**Comment by dmrvb** [comment]
I've got a file containing lots of Function definitions (each containing just a couple of lines of code). This takes 7 seconds to do the code hihglighting which is a bit irritating. Does the highlighting need to be rebuilt every time we switch between code editors?

**Comment by WaynePhillipsEA** [comment]
I'd like to take a look at these projects if possible, either on here, or privately.  If I can reproduce, then it will get fixed for sure.

**Comment by WaynePhillipsEA** [comment]
> Does the highlighting need to be rebuilt every time we switch between code editors?

That should've been addressed in BETA33.  Is it still a problem for you in the latest beta?

**Comment by greatquux** [comment]
[cpu_example.zip](https://github.com/WaynePhillipsEA/twinbasic/files/8673000/cpu_example.zip)
Here is a project file that contains the LEADTOOLS header file above that takes minutes to open up properly.

**Comment by dmrvb** [comment]
I've just installed 34 and its still doing a slow update everytime I switch to the editor.
The slowest module is "mIID" which is part of the lovely oleexp project from fafalone on VBforums. As its already in the public domain I hope I'm not breaking any rules by including it here
[mIID.bas.txt](https://github.com/WaynePhillipsEA/twinbasic/files/8673031/mIID.bas.txt)
.

**Comment by WaynePhillipsEA** [comment]
@greatquux the significant delay in opening `L_OCXADD.bas` is not due to the rendering of the semantic highlighting.  It's actually due to the creation of the DOM nodes for the OUTLINE panel.   If you turn all of the options off in the OUTLINE panel (use the ellipses button to access the menu), you'll find that the file will then open instantly.   I'll look to get this fixed today.

**Comment by WaynePhillipsEA** [comment]
Please check out BETA 35.   It should fix the problem for both of you.  The delay should now be much less noticeable.

**Comment by dmrvb** [comment]
Yes, thats much better, thanks. There is still a delay everytime that I switch to the relevant code editor window buts its about 1 second which is no problem.

**Comment by WaynePhillipsEA** [comment]
Great.  I think that's acceptable for now.  There's more we can do, but there are other priorities, so we'll come back to it.  Thanks.

I will await @greatquux to confirm it's fixed also before closing this issue.

**Comment by greatquux** [comment]
YES this is a lot better now.  If I leave all the DOM options checked, typing is still a bit slow, but I don't need any of them so I unchecked them, tried all sorts of files and typing a few test functions, and it's great.  It looks like those options do stay unchecked when opening and closing projects or the program.   One minor thing I noticed was that (sometimes not all the times) when you double click a file to open it from the project explorer, the file will open blank but if you click on another open tab and then back to the file it displays.  But that is minor and not really related to this so you can close this bug.
